### PR TITLE
Update to latest substrate-master and polkadot v0.3

### PR DESCRIPTION
**Pull request by bkchr**
This branch combines latest polkadot master, v0.3 and substrate master.

### CODE REVIEW

File: service/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "polkadot-service"
-version = "0.3.0"
```
Comment (andresilva) [inline service/Cargo.toml]: we should probably use a different version for master branch (maybe 0.4?).

File: network/src/router.rs [hunk 2]
```diff
@@ -254,7 +260,41 @@ impl<P: ProvideRuntimeApi + Send, E, N, T> TableRouter for Router<P, E, N, T> wh
...
 	fn drop(&mut self) {
-		self.fetcher.network().drop_gossip(self.attestation_topic);
+		let parent_hash = self.parent_hash().clone();
+		self.message_validator.remove_session(&parent_hash);
+		self.network().with_spec(move |spec, _| { spec.remove_validation_session(parent_hash); });
+	}
+}
+
+/// Receiver for block data.
+pub struct BlockDataReceiver {
```
Comment (andresilva) [inline network/src/router.rs]: This doesn't seem to be used anymore.

File: Cargo.toml [hunk 3]
```diff
@@ -4,7 +4,7 @@ path = "src/main.rs"
 
 [package]
 name = "polkadot"
-version = "0.3.0"
+version = "0.3.20"
```
Comment (andresilva) [inline Cargo.toml]: Bump to `0.4`?

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -254,15 +256,14 @@ construct_runtime!(
 		Staking: staking,
 		Democracy: democracy,
 		Grandpa: grandpa::{Module, Call, Storage, Config<T>, Log(), Event<T>},
+		CuratedGrandpa: curated_grandpa::{Module, Call, Config<T>, Storage},
```
Comment (rphmeier) [inline runtime/src/lib.rs]: If we are using `curated_grandpa`, we should remove `grandpa::SyncedAuthorities` from `OnSessionChange`.

**Pull request merged**